### PR TITLE
Added pipelining syntax for generators

### PR DESCRIPTION
**Pull request by worktycho**
Also converted part of cBioGenGrown as an example.

### CODE REVIEW

**Comment by madmaxoft** [comment]
I have to say, I don't like this code at all. It is extremely twisted, very hard to understand. And the original code requires the Underlying generator assignment within constructor, which the new code doesn't have.

**Comment by worktycho** [comment]
If you look, I haven't changed the constructors of the generators. Instead, I created a wrapper type cIntGenFactory to hold the args until we can construct the generator. As for the twistedness, it came out a lot more twisted than I expected and I am going to see if I can clean it.

**Comment by worktycho** [comment]
Also check basic style fails for `std::shared_ptr<Gen> construct(LhsGen&& lhs)`. This is a general problem for declarations using rvalue references and templates that do not contain `T` in them.

**Comment by worktycho** [comment]
I tried to simplify this then found a bug and ended up increasing the twistyness.

**Comment by madmaxoft** [comment]
I thought we agreed on changing the operator to pipe. As it is now, there's just way too many anglebrackets.

I suppose style checks are failing for you as well, you have curly braces at end of lines.

**Comment by worktycho** [comment]
I compleatly forgot about changing the operator change. The style checks are partially failing on brackets and partially on false positive.

**Comment by worktycho** [comment]
What do you think of the changes now?

File: src/Generating/IntGen.h [hunk 1]
```diff
@@ -62,9 +67,70 @@ class cIntGen
...
+
+// Code adapted from http://stackoverflow.com/questions/7858817/unpacking-a-tuple-to-call-a-matching-function-pointer
+
+template<int ...>
+struct seq
+{
+};
+
+template<int N, int ...S>
+struct gens : gens<N-1, N-1, S...>
```
Comment (madmaxoft) [inline src/Generating/IntGen.h]: Spaces around the operators

**Comment by madmaxoft** [comment]
Functionally, it's okay. Stylistically it's rather poor. Spaces are missing around most symbols, `a_CamelCase` is not used etc.

**Comment by worktycho** [comment]
I've fixed the style issues I can identify.

**Comment by madmaxoft** [comment]
Looks good enough, merge if it compiles.

**Pull request merged**